server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3844,7 +3848,10 @@ private UserVm createVirtualMachine(DataCenter zone, ServiceOffering serviceOffe
             // Check is hostName is RFC compliant
             checkNameForRFCCompliance(hostName);
         }
-        instanceName = VirtualMachineName.getVmName(id, owner.getId(), _instance);
+        instanceName = VirtualMachineName.getVmName(id, owner.getId(), instanceSuffix);
```

Comment:
here `instanceName` variable is used to set name on hypervisor but the same variable is used later for internal name.
`instanceSuffix` is assigned based on displayName (not considering instanceNameFlag), does this effect the case where flag is false and displayName is provided ? 
In that case internal-name should be `i - <user_id> - <vm_id> -<instance.name>` but here we are assigning `i - <user_id> - <vm_id> -<hostName>`.

Reply:
@harikrishna-patnala default value of `instanceSuffix` is `_instance` (which should get instance.name). Only in case of VMware and instanceFlag it is set to hostname at ln 3832.
Therefore, only when instanceFlag and other conditions are satisfied `instanceName` will be `i - <user_id> - <vm_id> -<hostName>` and will be later set to hypervisor

Reply:
Thanks @shwstppr I missed the check at 3823